geetools/ee_image_collection.py

```diff
@@ -896,6 +896,25 @@ def groupInterval(self, unit: str = "month", duration: int = 1) -> ee.List:
             lambda dr: ic.filterDate(ee.DateRange(dr).start(), ee.DateRange(dr).end())
         )
 
+        sizeName = "__geetools_generated_size__"
+
+        def add_size(ic):
+            ic = ee.ImageCollection(ic)
+            return ic.set({sizeName: ic.size()})
+
+        def delete_size_property(ic):
+            ic = ee.ImageCollection(ic)
+            propertiesToCopy = ic.propertyNames().remove(sizeName)
+            return ee.ImageCollection(ic.toList(ic.size())).copyProperties(
```

Comment:
why are you transforming it into a list here ? This is a very expensive step and i'm not sure it's necessary. 
```python 
return ee.imageCollection( # forced to recast because copyProeprty is a computed OBject 
    ic.copyProperties(ic, properties=propertiesToCopy)
)
```
can you switch the variable name to "toCopy" so everything can fit nicely on 1 line ?